assets/app/scripts/directives/nav.js

```diff
@@ -53,11 +53,16 @@ angular.module('openshiftConsole')
 
           sortedProjects = $filter('orderByDisplayName')(projects);
 
-          options = _.map(sortedProjects, function(item) {
+          options = _.map(sortedProjects, function(item, index) {
```

Comment:
What about a filter that takes a set of items and returns a map of name:uniqueDisplayName. That lets you calculate once and share code